### DISCUSSION TITLE
Get session duration from terminal reporter

### DISCUSSION THREAD

**Discussion by BeyondEvil** [pytest-dev/pytest#10147] [Q&A] [answered]
I'd like to retrieve the [session duration](https://github.com/pytest-dev/pytest/blob/34a285986e456406b22b62aa563a64331f640e7f/src/_pytest/terminal.py#L1044) for use in the pytest-html report.

Is that possible today?

From what I can tell, it's not available on the TerminalReporter object.

**Comment by The-Compiler** [comment, marked as answer]
Do you need the *exact* same time as the one printed by pytest? If not, you can probably just implement `pytest_sessionstart` and `pytest_sessionfinish` like the terminalreporter does (or perhaps a hookwrapper around `pytest_cmdline_main` if you just want one function?), and then use `time.time()` there to get the time yourself. Should only be off by whatever time the terminalreporter needs itself.

**Reply by BeyondEvil** [reply]
I don’t need the exact time.

But I was looking to see if the value was available to me so that I didn’t have to reimplement the same logic. ☺️

Lazy programmer and whatnot. 😅
